tests/aws/services/cloudformation/test_template_engine.py

```diff
@@ -243,6 +243,19 @@ def test_cfn_template_with_short_form_fn_sub(self, deploy_cfn_template):
         result = stack.outputs["Result"]
         assert result == "test"
 
+    @markers.aws.validated
+    def test_sub_number_type(self, deploy_cfn_template):
+        alarm_name = "alarm-test-latency-preemptive"
+        stack = deploy_cfn_template(
+            template_path=os.path.join(
+                os.path.dirname(__file__), "../../templates/sub_number_type.yml"
+            ),
+            parameters={"ResourceNamePrefix": alarm_name},
+        )
+
+        result = stack.outputs["Result"]
```

Comment:
I would prefer us to verify that a `Number` parameter is correctly supplied to the resource that's deployed, i.e. round-trip a `Number` type through a resource, as that's a more complete test that the deployment is correct.

Perhaps the alarm name could be built from a `Number` parameter, e.g.

```yaml
      RestLatencyPreemptiveAlarm:
        Type: AWS::CloudWatch::Alarm
        Properties:
          AlarmName: !Sub "my-alarm-${RestLatencyPreemptiveAlarmPeriod}"
          # rest of attributes...
```

then assert the name is taken from the input parameter.

What do you think?

Reply:
Yes thats a great idea, rather I would test these three params individually: 
- alarm name(which confirms that `AWS::CloudWatch::Alarm` is getting deployed correctly on localstack which wasn't the case earlier and would throw the error as mentioned in the PR description)
- integer number type 
- float number type